infra/gcp/ensure-conformance-storage.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to create a new "conformance" bucket in GCS.
+#
+# Each conformance bucket exists in its own GCP project, and is writable by a
+# dedicated googlegroup.
```

Comment:
Can you say more about what gets written here and what the naming conventions should be?

Reply:
conformance test results in a way testgrid can consume:
- each is in a "directory" with some monotonically increasing UUID (commonly unix epoch time for third party uploads)
  - they should contain a `junit.*\.xml` file with the structured test results
  - they should (optionally but commonly) contain a build-log.txt with the raw test output
  - they should contain two small json files `started.json` and `finished.json` with some metadata

https://github.com/kubernetes/test-infra/tree/master/gubernator#gcs-layout

Reply:
we have a small python tool in test-infra to consume the build log and junit and upload these.
prow will produce results like this.

Reply:
I meant "...in the comments" :)

nit: timestamp is not a UUID.

Reply:
> nit: timestamp is not a UUID.

it's true, on prow these are actual snowflake IDs, however third party test results typically use an uploader that just uses the start time in unix epoch, as running two runs at the same start time doesn't make sense for this use case.

Reply:
(and we don't expect people to deploy prow to produce these)

Reply:
s/UUID/run identifier/